### PR TITLE
feat: spl compression and decompression 

### PR DESCRIPTION
**Pull request by SwenSchaeferjohann**
from jorrit/feat-decompress-spl
rebased to latest main

### CODE REVIEW

File: programs/compressed-token/src/process_mint.rs [hunk 1]
```diff
@@ -281,6 +285,7 @@ pub mod mint_sdk {
             mint: *mint,
             mint_authority_pda,
             token_program: anchor_spl::token::ID,
+            cpi_authority_pda: get_cpi_authority_pda().0,
```
Comment (SwenSchaeferjohann) [inline programs/compressed-token/src/process_mint.rs:288]: this isnt used in CreateMintInstruction

File: js/stateless.js/src/test-utils/test-rpc.ts [hunk 2]
```diff
@@ -270,10 +270,12 @@ export class TestRpc extends Rpc {
             allLeaves.map(leaf => bn(leaf).toString()),
         );
 
+        console;
```
Comment (ananas-block) [inline js/stateless.js/src/test-utils/test-rpc.ts:273]: leftover?

File: programs/compressed-token/src/process_mint.rs [hunk 3]
```diff
@@ -35,6 +35,9 @@ pub struct CreateMintInstruction<'info> {
     #[account(mut, seeds=[MINT_AUTHORITY_SEED, authority.key().to_bytes().as_slice(), mint.key().to_bytes().as_slice()], bump)]
     pub mint_authority_pda: AccountInfo<'info>,
     pub token_program: Program<'info, Token>,
+    /// CHECK: TODO: FIXME: unsafe
```
Comment (ananas-block) [inline programs/compressed-token/src/process_mint.rs:38]: it's safe this was just an anchor warning

File: programs/compressed-token/src/process_mint.rs [hunk 4]
```diff
@@ -262,14 +265,15 @@ pub mod mint_sdk {
...
-    use crate::{get_token_authority_pda, get_token_pool_pda};
+    use crate::{get_cpi_authority_pda, get_token_authority_pda, get_token_pool_pda};
 
     pub fn create_initialize_mint_instruction(
         fee_payer: &Pubkey,
         authority: &Pubkey,
         mint: &Pubkey,
     ) -> Instruction {
         let token_pool_pda = get_token_pool_pda(mint);
+        println!("token_pool_pda: {:?}", token_pool_pda);
```
Comment (ananas-block) [inline programs/compressed-token/src/process_mint.rs:276]: leftover print?

**Pull request closed without merging**